File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        ans = 0
        for num in numSet:
            cnt = 1
            if (num-1) not in numSet:
                while (num + cnt) in numSet:
                    cnt = cnt + 1
                ans  =max(ans, cnt)
        return ans

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int>m;

        int ans = 0, l = 0;
        int n = s.size();
        for(int r = 0; r < n; r++){
            if(m.find(s[r]) != m.end()) {
                l = max(l,m[s[r]] + 1);
            }
            m[s[r]] = r;
            ans = max(ans, r-l + 1);
        }
        return ans;
    }
};
